scripts/osx/testTemplate.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+OF_DIR=`pwd`;
+echo $OF_DIR;
+TEST_DIR="$OF_DIR/scripts/osx"
+cd $TEST_DIR
```

Comment:
It can even be more clear/concise like this:

```
matrix:
  - NAME="osx" SCRIPT="testTemplate.sh"
  - NAME="ios" SCRIPT="testTemplate.sh"
  - NAME="doc" SCRIPT="documentation.sh"
before_install:
  "if [ \"$NAME\" == \"doc\" ]; then scripts/$NAME/$SCRIPT before_install; fi"
script:
  scripts/$NAME/$SCRIPT
after_success:
  "if [ \"$NAME\" == \"doc\" ]; then scripts/$NAME/$SCRIPT after_success; fi"
```
the only necessary change is to move the documentation script into a scripts/doc subfolder, which is probably a reasonable thing to do anyway.
Script is not tested since I can't do that right now. Also, I think you can get rid of most of those quotes and escaped quotes, no? (from looking at other travis.yml files)